azure_adx/main.tf

```diff
@@ -0,0 +1,63 @@
+resource "azurerm_resource_group" "adx" {
+  name     = var.resource_group_name
+  location = var.location
+}
+
+resource "azurerm_kusto_cluster" "adx" {
+  name                        = var.adx_cluster_name
+  location                    = azurerm_resource_group.adx.location
+  resource_group_name         = azurerm_resource_group.adx.name
+  disk_encryption_enabled     = true
+  streaming_ingestion_enabled = true
+
+  sku {
+    name     = var.adx_sku
+    capacity = var.adx_capacity
+  }
+
+}
+resource "azurerm_kusto_database" "adx" {
+  name                = var.adx_cluster_name
+  resource_group_name = azurerm_resource_group.adx.name
+  location            = azurerm_resource_group.adx.location
+  cluster_name        = azurerm_kusto_cluster.adx.name
+
+  hot_cache_period   = "P7D"
```

Comment:
I would suggest changing these to use variables, and then set the defaults in the variable, to provide maximum flexibility.